src/containers/sound-editor.jsx

```diff
@@ -322,25 +327,37 @@ class SoundEditor extends React.Component {
         });
     }
     resampleBufferToRate (buffer, newRate) {
-        return new Promise(resolve => {
+        return new Promise((resolve, reject) => {
+            const sampleRateRatio = newRate / buffer.sampleRate;
+            const newLength = sampleRateRatio * buffer.samples.length;
+            let offlineContext;
             if (window.OfflineAudioContext) {
-                const sampleRateRatio = newRate / buffer.sampleRate;
-                const newLength = sampleRateRatio * buffer.samples.length;
-                const offlineContext = new window.OfflineAudioContext(1, newLength, newRate);
-                const source = offlineContext.createBufferSource();
-                const audioBuffer = offlineContext.createBuffer(1, buffer.samples.length, buffer.sampleRate);
-                audioBuffer.getChannelData(0).set(buffer.samples);
-                source.buffer = audioBuffer;
-                source.connect(offlineContext.destination);
-                source.start();
-                offlineContext.startRendering();
-                offlineContext.oncomplete = ({renderedBuffer}) => {
-                    resolve({
-                        samples: renderedBuffer.getChannelData(0),
-                        sampleRate: newRate
-                    });
-                };
+                offlineContext = new window.OfflineAudioContext(1, newLength, newRate);
```

Comment:
I don't understand what cases are being covered by these two different ways of getting offline audio context.

Reply:
`OfflineAudioContext` is not supported on all browsers we support. Specifically, safari still has the offline audio context under vendor prefix.

Reply:
maybe add a comment before this test explaining that -- something like "there are two ways major browsers support resampling: window.OfflineAudioContext and window.webkitOfflineAudioContext"

and maybe before the line `offlineContext = new window.OfflineAudioContext(1, newLength, newRate);` you could comment "if window.OfflineAudioContext exists, we're confident we can use it to resample"

and before the `offlineContext = new window.webkitOfflineAudioContext(1, newLength, newRate);` line below you might comment "some webkit browsers are currently known to crash on some sample rates"